src/OrleansSQLUtils/Storage/DbStoredQueries.cs

```diff
@@ -0,0 +1,425 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Net;
+using System.Reflection;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.SqlUtils;
+
+namespace OrleansSQLUtils.Storage
+{
+    /// <summary>
+    /// This class implements the expected contract between Orleans and the underlying relational storage.
+    /// It makes sure all the stored queries are present and 
+    /// </summary>
+    internal class DbStoredQueries
+    {
+        private readonly Dictionary<string, string> queries;
+        internal DbStoredQueries(Dictionary<string, string> queries)
+        {
+            var fields = typeof (DbStoredQueries).GetProperties(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Select(p => p.Name);
+            var missingQueryKeys = queries.Keys.Except(fields).ToArray();
+            if (missingQueryKeys.Length > 0)
+            {
+                throw new ArgumentException(
+                    $"Not all required queries found. Missing are: {string.Join(",", missingQueryKeys)}");
+            }
+            this.queries = queries;
+        }
+
+        /// <summary>
+        /// the query that's used to get all the stored queries.
```

Comment:
Ah, "the query that's used to get all the stored queries.". Most (if not all, I'm not sure) comments are capitalized and end with a full stop. I don't know if there's a guidance for this in Orleans. So as to be clear, I go mainly how others go, but I prefer full sentences using proper writing (I explain it to myself it shouldn't at least hurt for trying to be concise and use proper language).